src/ccl/auto-complete/auto-complete.component.ts

```diff
@@ -0,0 +1,233 @@
+import {Component, ElementRef, Input, OnInit, ViewEncapsulation} from '@angular/core';
+import {Subject} from "rxjs/Subject";
+import {AutoComplete} from './auto-complete';
+
+var module: any; // just to pass type check
+/**
+ * show a selected date in monthly calendar
+ * Each filteredList item has the following property in addition to data itself
+ *   1. displayValue as string e.g. Allen Kim
+ *   2. dataValue as any e.g. 1234
+ */
+@Component({
+  selector: 'auto-complete',
+  template: `
+  <div class="auto-complete">
+
+    <!-- keyword input -->
+    <input class="keyword"
+           placeholder="{{placeholder}}"
+           (focus)="showDropdownList()"
+           (blur)="dropdownVisible=false"
+           (keydown)="inputElKeyHandler($event)"
+           (input)="reloadListInDelay()"
+           [(ngModel)]="keyword" />
+
+    <!-- dropdown that user can select -->
+    <ul *ngIf="dropdownVisible">
+      <li *ngIf="isLoading" class="loading">Loading</li>
+      <li class="item"
+          *ngFor="let item of filteredList; let i=index"
+          (mousedown)="selectOne(item)"
+          [ngClass]="{selected: i === itemIndex}"
+          [innerHTML]="getFormattedList(item)"
+          ></li>
+    </ul>
+
+  </div>`,
+  providers: [ AutoComplete ],
+  styles: [`
+  @keyframes slideDown {
```

Comment:
Considering this came directly from a copied project I think it makes sense to preserve it to be able to do diffs.